### PR TITLE
Add ability to use ubuntu base images.

### PR DESCRIPTION
**Pull request by mkalcok**
This change adds ability to use Ubuntu base image by introducing platform-specific install scripts that are conditionally loaded during the `cinc` image build. The install script is chosen based in the value of `$OS_BASE` env. variable.

The change is backward compatible and maintains the default behavior of building on top of `fedora:latest` image.

I intend to open a follow up PR in `ovn-heater` that would enable running test scenarios either on `fedora` or `ubuntu` based images.

### CODE REVIEW

File: image/cinc/install_fedora_pkg.sh [hunk 1]
```diff
@@ -53,6 +53,7 @@ dnf -y --skip-broken install \
   python3-devel \
   python3-pip \
   python3-psutil \
+  python3-six \
```
Comment (fnordahl) [inline image/cinc/install_fedora_pkg.sh:56]: This changes the behavior of the fedora build.

I understand that you want to move the `pip3 -qq install six` away from the common part above, but is it desired/necessary to change the behavior for the existing OS support? If so, should it perhaps be a parent commit with an explanation?

File: image/cinc/install_ubuntu_pkg.sh [hunk 2]
```diff
@@ -0,0 +1,52 @@
+#!/bin/bash -xe
+
+DEBIAN_FRONTEND=noninteractive
+
```
Comment (fnordahl) [inline image/cinc/install_ubuntu_pkg.sh:4]: The fedora install script contains a section which masks several services.

Do we perhaps need to mask some services in the Ubuntu build as well? From cursory view of the CI log I see errors:
https://github.com/ovn-org/ovn-fake-multinode/actions/runs/6695210331/job/18190262467?pr=85#step:5:500

**Review by fnordahl — CHANGES_REQUESTED** [review]
Thanks for working on this, Martin!

This generally looks good to me, and I have a couple of comments in-line that needs attention.

File: image/cinc/install_fedora_pkg.sh (hunk 1 above)
Comment (mkalcok) [reply image/cinc/install_fedora_pkg.sh:56]: I saw this as an inconsistency and I just thought I'd fix it. Other python packages are installed in install script using `dnf`/`apt`, it's just this one that's installed explicitly in the Dockerfile. The package versions match in all three sources `pypi`/`apt`/`dnf` for current releases and I did not observe any behavior change. But you are right, perhaps I should move it to a separate commit.

File: image/cinc/install_ubuntu_pkg.sh (hunk 2 above)
Comment (mkalcok) [reply image/cinc/install_ubuntu_pkg.sh:4]: Thanks for noticing. I though that I'm skipping `systemd` installation, but it seems that it's a dependency for `init` package. I'll add the service masking.

**Comment by mkalcok** [comment]
Aaaaand I messed up the interactive rebase and brought in unintended commits :D

**Review by fnordahl — APPROVED** [review]
LGTM, thanks!

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -7,12 +7,23 @@ on:
 
 jobs:
   basic:
+    strategy:
+      matrix:
+        image:
+          # Default fedora base image
+          - url: ""
+            base: ""
```
Comment (almusil) [inline .github/workflows/ci.yml:15]: We should be explicit here, even though the Fedora is default when nothing is specified, at first glance it's not clear which Fedora version is used.

File: install_ovn.sh [hunk 4]
```diff
@@ -18,6 +18,8 @@ set -o errexit
 use_ovn_rpm=$1
 extra_optimize=$2
 
+export PIP_BREAK_SYSTEM_PACKAGES=1
```
Comment (almusil) [inline install_ovn.sh:21]: Can you please add small comment why is it needed now and wasn't before?

File: .github/workflows/ci.yml (hunk 3 above)
Comment (mkalcok) [reply .github/workflows/ci.yml:15]: My intention here was to test that the previous default behavior (i.e. no env vars exported) keeps working. Would it help to put in more detailed comment?

Comment (almusil) [reply .github/workflows/ci.yml:15]: I see, in that case please add more detailed comment about the intention and that the default is `fedora:latest`.

**Comment by mkalcok** [comment]
Thanks for the review @almusil . I updated the PR based on your comments.

**Review by almusil — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**